lib/ably/auth.rb

```diff
@@ -482,15 +486,16 @@ def can_assume_client_id?(assumed_client_id)
     #
     # @api private
     def configure_client_id(new_client_id)
-      # If new client ID from Ably is a wildcard, but preconfigured clientId is set, then keep the existing clientId
-      if has_client_id? && new_client_id == '*'
-        @client_id_validated = true
-        return
-      end
-
-      # If client_id is defined and not a wildcard, prevent it changing, this is not supported
-      if client_id && client_id != '*' &&  new_client_id != client_id
-        raise Ably::Exceptions::IncompatibleClientId.new("Client ID is immutable once configured for a client. Client ID cannot be changed to '#{new_client_id}'")
+      if has_client_id?
```

Comment:
What is the purpose of this change? Doesn't seem related to the rest of the commit.

Reply:
Actually, there was a repetitive logic, so it's a small refactoring to better understand conditional checks

Reply:
OK, I see. It would have been better to put this in a separate commit to avoid confusion.